### PR TITLE
Create a release drafter configuration

### PR DESCRIPTION
**Pull request by oleg-nenashev**
It is inspired by https://github.com/jenkinsci/configuration-as-code-plugin/blob/master/.github/release-drafter.yml from @casz . It contains few minor tweaks for Jenkinsfile Runner, but generally we can evolve it to an organization-wide configuration sample which may be helpful for repositories which use GitHub releases or CHANGELOG.md

Example: https://github.com/oleg-nenashev/jenkinsfile-runner/releases (scroll down through uncategorized changes)

Sample:

JFR+Changelog_sample

### CODE REVIEW

**Review by jetersen — APPROVED** [review]

**Comment by oleg-nenashev** [comment]
CI fails due to whatever reason:

```
[2019-05-19T10:01:06.841Z] [ERROR] Failed to execute goal org.codehaus.mojo:exec-maven-plugin:1.6.0:exec (test) on project tests: Execution test of goal org.codehaus.mojo:exec-maven-plugin:1.6.0:exec failed: Plugin org.codehaus.mojo:exec-maven-plugin:1.6.0 or one of its dependencies could not be resolved: Failed to collect dependencies at org.codehaus.mojo:exec-maven-plugin:jar:1.6.0 -> org.apache.commons:commons-exec:jar:1.3: Failed to read artifact descriptor for org.apache.commons:commons-exec:jar:1.3: Could not transfer artifact org.apache.commons:commons-exec:pom:1.3 from/to repo.jenkins-ci.org (https://repo.jenkins-ci.org/public/): /home/jenkins/.m2/repository/org/apache/commons/commons-exec/1.3/commons-exec-1.3.pom.part.lock (No such file or directory) -> [Help 1]

```

File: .github/release-drafter.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  - no-changelog
+  - skip-changelog
+  - invalid
+
+template: |
+  ## Changes
+  $CHANGES
+
+replacers:
+  - search: '/\[*JENKINS-(\d+)\]*\s*-*\s*/g'
```
Comment (jetersen) [inline .github/release-drafter.yml:33]: Gotta be careful these might not always match, depending on how users write their PR titles.

Comment (oleg-nenashev) [reply .github/release-drafter.yml:33]: Right. We just need to start from something. I can also add a straightforward `JENKINS-123` replacement

File: .github/release-drafter.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+# Emoji reference: https://gitmoji.carloscuesta.me/
+categories:
+  - title: 🚨 Removed
+    label: removed
+  - title: ⚠️ Deprecated
+    label: deprecated
+  - title: 🚀 New features and improvements
+    label: enhancement
+  - title: 🐛 Bug Fixes
+    label: bugfix
```
Comment (jetersen) [inline .github/release-drafter.yml:15]: the default label is `bug` we went with `fix` but `bugfix` seems fine :sweat_smile:

Comment (oleg-nenashev) [reply .github/release-drafter.yml:15]: `bug` always raises a question whether it is "added a new bug" or "fixed a bug" :)

**Review by fcojfernandez — APPROVED** [review]
Really nice feature!

**Comment by oleg-nenashev** [comment]
Next steps is to have a default configuration on the org-wide basis. I will create a jenkinsci-dev thread for it

**Pull request merged**